### ISSUE TITLE
Carbon Reduction/Absorption error handling

### ISSUE THREAD

**Issue by piotroslaw** [farhanrahman/kyoto#27]
If you look at both handlers, they are a bit messy when it comes to errors. They may throw NotEnoughXXX errors, which is good. The problem is that their behaviour in case of any other error is uneven:
- some functions use try catch blocks and return 0 or MAX_VALUE in case of error
- some don't use try catch blocks at all, and leave it to countries to catch any errors

The issue with the former is that we don't have any feedback on errors. since handlers don't have access to countries' loggers, they have no way of indicating that something went wrong - they can only return some ridiculous variable.

So my proposition is to go with the latter in every function, i.e. do not catch any errors and adjust return variables in the handlers, leaving it to the countries to catch any errors and decide what to do with erroneous reutrn variable.

I am happy to do the cleaning up in both handlers if everyone agrees

**Comment by piotroslaw** [comment]
tl;dr version:

I think it's better to catch exceptions where they can be properly logged and handled in uniform way, i.e. in countries using a handler

**Comment by cmd08** [comment]
Might be best to re-throw them as a different type of exception, but in general you are 100% correct. Please close this issue (and associate a commit to it if you can) when you're done :)

**Comment by piotroslaw** [comment]
It turned out that some changes were already made, but were not pushed to development. I took your (Chris) comment into account, and changed it so that now funtions catch exceptions (to know where they occured exactly, needed for debugging etc - this was partially implemented in the meantime by printing to console), but then throws exceptions "up". At each stage, the new function name is added to error message, so that we have both advantages:
- you can see exactly the error "trace"
- you catch the errors in country code, so that a proper action can be taken.

I'm closing this issue for now, but if anything's wrong just write here or change the file.
